### ISSUE TITLE
Need guidance on complicated folder structure (Serverless Framework)

### ISSUE THREAD

**Issue by ffxsam** [getsentry/sentry-webpack-plugin#255]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
Node.js, 6.0.2

I'm using the Serverless Framework to deploy AWS Lambda functions, and I'm using the `package.individually: true` option in my serverless.ts config file, which basically splits out each Lambda function in order to reduce the amount of duplicated code among all functions.

To elaborate a bit: basically if I have Function A and Function B, with `package.individually: false`, the deployed functions would all have the same bundle size. This is what the `.webpack` folder structure looks like:

<img width="250" alt="Screen Shot 2021-01-29 at 10 47 29 AM" src="https://user-images.githubusercontent.com/12532733/106303600-f04cb800-621f-11eb-9af3-8a1e357684ea.png">

This is pretty straightforward, and all I have to do for the Sentry webpack plugin is to set `include: './.webpack/service/src'` in webpack.config.js.

However, if I set `package.individually: true`, the functions are bundled separately. So Function A might be 8MB but Function B only be 20KB (as opposed to the first option where all functions would be 8MB).

That `.webpack` folder structure looks like this:

<img width="250" alt="Screen Shot 2021-01-29 at 10 47 40 AM" src="https://user-images.githubusercontent.com/12532733/106303799-2c801880-6220-11eb-9a6c-60337b7e6f59.png">

How can I configure `SentryCliPlugin` to work with this folder structure?

**Comment by ffxsam** [comment]
I'd also like to add that this plugin doesn't work out of the box for me (with Serverless + AWS Lambda). I had to add this code to my Sentry integrations:

```js
    new RewriteFrames({
      iteratee: frame => {
        if (/^\/var\/task\/src\//.test(frame.filename)) {
          frame.filename = frame.filename.replace(
            /^\/var\/task\/src\//,
            'app:///'
          );
        }
        return frame;
      },
    }),
```

**Comment by ffxsam** [comment]
Ok, after many hours of trial & error, I think I've solved this. If this can in any way be improved upon, please weigh in!

## webpack.config.js changes

I didn't have to change a lot here. But this part was absolutely key:
```js
module.exports = {
  output: {
    // other config
    devtoolModuleFilenameTemplate: 'app:///[resource-path]',
  },
};
```

Sentry didn't like seeing `webpack://` as the prefix, so I had to change the config to use `app:///`.

And since I'm setting `package.individually: true` in serverless.ts, I've set the Sentry plugin to include the whole `.webpack` folder:

```js
    new SentryCliPlugin({
      org: 'reelcrafter',
      project: 'v2-media-processor',
      include: './.webpack',
    }),
```

## Sentry.init changes

This took a little bit of guesswork and hacking for the `RewriteFrames` options:
```js
    new RewriteFrames({
      iteratee: frame => {
        if (/^\/var\/task\/src\//.test(frame.filename)) {
          frame.filename = frame.filename.replace(
            /^\/var\/task\/src\//,
            'app:///' + camelCase(__dirname.split('/').slice(-1)[0]) + '/src/'
          );
        }
        console.log('updated frame:', frame);
        return frame;
      },
    }),
```

It makes some assumptions as to folder structure, but basically, each of my Lambda functions resides in `<project root>/src/functions/my-function-name` so this works fine.

And now, when I throw an error:

<img width="811" alt="image" src="https://user-images.githubusercontent.com/12532733/106365457-6114d280-62fb-11eb-9f57-304aeaa3dc0e.png">

The `null.<anonymous>` is a bit weird, but not a concern as long as it shows me the filename and line number.

**Comment by ajjindal** [comment]
@ffxsam As we discussed, looks like you have figured out a solution for now. Can you please confirm?

**Comment by ffxsam** [comment]
Yes I have! But if there's a more elegant way than what I've come up with, please let me know. Thanks.

**Comment by kamilogorek** [comment]
Closing the issue as a part of large repository cleanup, due to it being inactive and/or outdated.
Please do not hesitate to ping me if it is still relevant, and I will happily reopen it.
Cheers!
